### ISSUE TITLE
Workers hanging after task completion

### ISSUE THREAD

**Issue by fsiddi** [fsiddi/brender#82]
This is critical, and is preventing the brender to be run in production at the moment. At the end of a task, sometimes the worker hangs, preventing the manager to update, which in turn prevents the server to update.

By pressing Ctrl+C once in the worker terminal, I get this:

```
[DEBUG] return code: 0
^C----------------------------------------
Exception happened during processing of request from ('127.0.0.1', 58901)
Traceback (most recent call last):
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/SocketServer.py", line 295, in _handle_request_noblock
    self.process_request(request, client_address)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/SocketServer.py", line 321, in process_request
    self.finish_request(request, client_address)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/SocketServer.py", line 334, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/SocketServer.py", line 649, in __init__
    self.handle()
  File "/Users/fsiddi/Documents/brender/venv/lib/python2.7/site-packages/werkzeug/serving.py", line 200, in handle
    rv = BaseHTTPRequestHandler.handle(self)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/BaseHTTPServer.py", line 340, in handle
    self.handle_one_request()
  File "/Users/fsiddi/Documents/brender/venv/lib/python2.7/site-packages/werkzeug/serving.py", line 235, in handle_one_request
    return self.run_wsgi()
  File "/Users/fsiddi/Documents/brender/venv/lib/python2.7/site-packages/werkzeug/serving.py", line 177, in run_wsgi
    execute(self.server.app)
  File "/Users/fsiddi/Documents/brender/venv/lib/python2.7/site-packages/werkzeug/serving.py", line 165, in execute
    application_iter = app(environ, start_response)
  File "/Users/fsiddi/Documents/brender/venv/lib/python2.7/site-packages/flask/app.py", line 1836, in __call__
    return self.wsgi_app(environ, start_response)
  File "/Users/fsiddi/Documents/brender/venv/lib/python2.7/site-packages/flask/app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/fsiddi/Documents/brender/venv/lib/python2.7/site-packages/flask/app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/fsiddi/Documents/brender/venv/lib/python2.7/site-packages/flask/app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/fsiddi/Developer/brender/brender/worker/application/controllers.py", line 242, in execute_task
    time.sleep(1)
KeyboardInterrupt
----------------------------------------
```

Pressing Ctrl+C a second time stops the worker. I'll post here any further investigation result I come across.

**Comment by fsiddi** [comment]
I noticed that if the number of frames is 20, and the number of chunks is 5 (making 4 full tasks), the file renders fine. If the frames are 22, then the worker hangs.

**Comment by fsiddi** [comment]
So, as you said, it really seems to be a problem related to the fact that if completing a task takes less than 1 second, the worker is left hanging.

**Comment by msimoncz** [comment]
It is because the worker sends the PATCH request before answering to execute_task

**Comment by Eibriel** [comment]
Solved on d274c46ad2b31431c4178549913d2d23f147f0d5
